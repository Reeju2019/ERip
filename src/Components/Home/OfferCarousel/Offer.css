.Offer-Carousel {
  background-color: #ffffff;
}
.slick-arrow {
  color: blue !important;
}

.carousel-content-1 {
  margin: 0 12%;
}

/* How it works section */

.side-gap {
  margin: 0 10%;
}

.work-row {
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0 13%;
}

.first-work {
  margin-top: 4%;
  margin-bottom: 4%;
  padding-left: 3%;
}

.second-work {
  margin-top: 4%;
  margin-bottom: 4%;
  padding-left: 3%;
}

.works-img-pick {
  width: 13%;
  height: 0%;
  margin-right: 2%;
}

.works-img-service {
  width: 13%;
  height: 0%;
  margin-right: 2%;
}

.work-heading {
  font-size: 1.1rem;
  font-weight: 600;
}

.work-para {
  color: #646464;
  font-size: 0.8125rem;
}

.middle-border {
  display: flex;
  justify-content: center;
  align-items: center;
}

hr {
  margin-bottom: 1rem;
}

.hr-mid-blue {
  border: 2px solid #2c80ff;
  width: 50%;
  margin-top: 7%;
  opacity: 1;
}
