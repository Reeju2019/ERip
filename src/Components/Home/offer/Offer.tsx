import React from 'react'
import './offer.css'
import data from '../../../Data/const.data.json'

const Offer = () => {
  return (
    <div>
      <section className='offers'>
        <div className='container-fluid offer-container'>
          {data.offer.map((e, index) => {
            return (
              <div key={index} className='row offer-2 d-flex justify-content-between'>
                <div className='col-7 col-sm-7'>
                  <h1 className='offer-text'>
                    <strong>
                      {e.heading1} <br /> {e.heading2}
                    </strong>
                  </h1>
                </div>
                <div className='offer-img col-4 col-sm-4 d-flex justify-content-end'>
                  <img src={e.img} className='image' alt='' />
                </div>
              </div>
            )
          })}
        </div>
      </section>
    </div>
  )
}

export default Offer
