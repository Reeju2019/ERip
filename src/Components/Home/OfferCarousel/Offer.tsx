import './Offer.css'
import Slider from 'react-slick'
import 'slick-carousel/slick/slick.css'
import 'slick-carousel/slick/slick-theme.css'
import { BsArrowRightShort, BsArrowLeftShort } from 'react-icons/bs'
import { Container, Image, Row } from 'react-bootstrap'
import OfferData from '../../../Data/OfferData.mock.json'

const Offer: React.FunctionComponent = () => {
  const settings = {
    dots: true,
    infinite: true,
    speed: 500,
    slidesToShow: 3,
    slidesToScroll: 1,
    prevArrow: <BsArrowLeftShort />,
    nextArrow: <BsArrowRightShort />,
    initialSlide: 0,
    responsive: [
      {
        breakpoint: 1024,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 1,
          infinite: true,
          dots: true,
        },
      },
      {
        breakpoint: 600,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1,
          initialSlide: 1,
        },
      },
      {
        breakpoint: 480,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1,
        },
      },
    ],
  }
  return (
    <>
      <section className='Offer-Carousel mt-5'>
        <div className='container'>
          <div className='row carousel-content-1'>
            <Slider {...settings}>
              <div className='carousel-item active justify-content-center'>
                <Image
                  src='static/Offer_carousel/Offer1.png'
                  className='d-block h-100 w-auto'
                  alt='...'
                />
              </div>

              <div className='carousel-item'>
                <Image
                  src='static/Offer_carousel/Offer2.png'
                  className='d-block h-100 w-auto'
                  alt='...'
                />
              </div>

              <div className='carousel-item'>
                <Image
                  src='static/Offer_carousel/Offer3.png'
                  className='d-block h-100 w-auto'
                  alt='...'
                />
              </div>
              <div className='carousel-item'>
                <Image
                  src='static/Offer_carousel/Offer4.png'
                  className='d-block h-100 w-auto'
                  alt='...'
                />
              </div>
              <div className='carousel-item'>
                <Image
                  src='static/Offer_carousel/Offer5.png'
                  className='d-block h-100 w-auto'
                  alt='...'
                />
              </div>
              <div className='carousel-item'>
                <Image
                  src='static/Offer_carousel/Offer6.png'
                  className='d-block h-100 w-auto'
                  alt='...'
                />
              </div>

              <div className='carousel-item'>
                <Image
                  src='static/Offer_carousel/Offer7.png'
                  className='d-block h-100 w-auto'
                  alt='...'
                />
              </div>

              <div className='carousel-item'>
                <Image
                  src='static/Offer_carousel/Offer8.png'
                  className='d-block h-100 w-auto'
                  alt='...'
                />
              </div>

              <div className='carousel-item'>
                <Image
                  src='static/Offer_carousel/Offer9.png'
                  className='d-block h-100 w-auto'
                  alt='...'
                />
              </div>
            </Slider>
          </div>
        </div>
      </section>
      <section className='vaccinated'>
        <Container>
          <Row>
            <div className='caro-2 d-flex'>
              <div className='text-vaccinated d-flex justify-content-start col-8'>
                <div className='d-flex flex-column'>
                  <p className='vaccinated-para d-flex justify-content-start m-0'>
                    {OfferData.vaccinPara}
                  </p>
                  <h1 className='vaccinated-heading '>
                    <strong className='d-flex justify-content-start'>
                      {OfferData.vaccinText1}
                    </strong>
                    <strong className='d-flex justify-content-start'>
                      {OfferData.vaccinText2}
                    </strong>
                  </h1>
                </div>
              </div>

              <Image
                src='static/Offer_carousel/img-fleet.png'
                className='d-flex justify-content-end col-4 vaccinated-img'
                alt=''
              />
            </div>
          </Row>
        </Container>
      </section>
      <section className='works'>
        <Container>
          <Row>
            <div className='col-12'>
              <h4 className='text-center mt-5'>
                <strong>{OfferData.workTitle}</strong>
              </h4>
            </div>
          </Row>
          <div className='row side-gap work-row d-flex '>
            <div className='first-work d-flex justify-content-center col-5 g-0'>
              <Image src='static/image/pick.png' className='works-img-pick' alt='' />
              <div className='work-text'>
                <h4 className='work-heading'>{OfferData.workSubtitle1}</h4>
                <p className='work-para'>
                  {OfferData.workSubtitle1Text1} <br /> {OfferData.workSubtitle1Text2}
                  <br /> {OfferData.workSubtitle1Text3}
                </p>
              </div>
            </div>

            <div className='middle-border col-2 g-0'>
              <hr className='hr-mid-blue' />
            </div>

            <div className='second-work d-flex justify-content-center col-5 g-0'>
              <Image src='static/image/Service.png' className='works-img-service' alt='' />
              <div className='work-test-1'>
                <h4 className='work-heading'>{OfferData.workSubtitle2}</h4>
                <p className='work-para'>
                  {OfferData.WorkSubtitle2Text1} <br /> {OfferData.workSubtitle2Text2} <br />
                  {OfferData.workSubtitle2Text3}
                </p>
              </div>
            </div>
          </div>
        </Container>
      </section>
    </>
  )
}

export default Offer
