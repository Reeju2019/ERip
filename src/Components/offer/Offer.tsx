import React from "react";
import "./offer.css";
import data from "../../Data/const.data.json";

const Offer = () => {
  return (
    <div>
      <section className="offers">
        <div className="container-fluid offer-container">
          {data.offer.map((e: any) => {
            return (
              <div className="row offer-2">
                <div className=" col-7 col-sm-8">
                  <h1 className="offer-text">
                    <strong>
                      {e.heading1} <br /> {e.heading2}
                    </strong>
                  </h1>
                </div>
                <div className="offer-img col-4 col-sm-4">
                  <img src={e.img} alt="" />
                </div>
              </div>
            );
          })}
        </div>
      </section>
    </div>
  );
};

export default Offer;
