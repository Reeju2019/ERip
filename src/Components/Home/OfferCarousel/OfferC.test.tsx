import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import Offer from './Offer'

test('offer c', () => {
  render(<Offer />)
})

describe('pnf rendering specification', () => {
  it('SnapShot testing', () => {
    const component = renderer.create(<Offer />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('render without crashing', () => {
    const div = document.createElement('div')
    render(<Offer />)
    ReactDOM.unmountComponentAtNode(div)
  })
})
