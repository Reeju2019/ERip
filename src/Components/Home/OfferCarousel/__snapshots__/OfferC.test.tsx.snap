// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pnf rendering specification SnapShot testing 1`] = `
Array [
  <section
    className="Offer-Carousel m-5"
  >
    <div
      className="container"
    >
      <div
        className="row carousel-content-1"
      >
        <div
          className="react-multi-carousel-list carousel-container "
          dir="ltr"
        >
          <ul
            className="react-multi-carousel-track "
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "overflow": "unset",
                "transform": "translate3d(0px,0,0)",
                "transition": "none",
              }
            }
          />
        </div>
      </div>
    </div>
  </section>,
  <section
    className="vaccinated"
    style={
      Object {
        "backgroundImage": "url(/static/image/bg-fleet.png)",
      }
    }
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="caro-2 d-flex"
        >
          <div
            className="text-vaccinated d-flex justify-content-start col-8"
          >
            <div
              className="d-flex flex-column"
            >
              <p
                className="vaccinated-para d-flex justify-content-start m-0"
              >
                #WeCare
              </p>
              <h1
                className="vaccinated-heading "
              >
                <strong
                  className="d-flex justify-content-start"
                >
                  Our Fleet is
                </strong>
                <strong
                  className="d-flex justify-content-start"
                >
                  Fully Vaccinated!
                </strong>
              </h1>
            </div>
          </div>
          <div
            className="pre_img"
          >
            <img
              alt="vaccine img"
              className="d-flex justify-content-end col-4 vaccinated-img"
              src="/static/Offer_carousel/img-fleet.png"
            />
          </div>
        </div>
      </div>
    </div>
  </section>,
  <section
    className="works"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <h4
            className="text-center mt-5"
          >
            <strong>
              How it Works
            </strong>
          </h4>
        </div>
      </div>
      <div
        className="row side-gap work-row d-flex "
      >
        <div
          className="first-work d-flex justify-content-center col-5 g-0"
        >
          <img
            alt=""
            className="works-img-pick"
            src="static/image/pick.png"
          />
          <div
            className="work-text"
          >
            <h4
              className="work-heading"
            >
              Pick Your Product
            </h4>
            <p
              className="work-para"
            >
              Visit our website, choose the product that needs
               
              <br />
               
              fixing, get a quote and Yay! There we are, right at
              <br />
               
               your doorstep.
            </p>
          </div>
        </div>
        <div
          className="middle-border col-2 g-0"
        >
          <hr
            className="hr-mid-blue"
          />
        </div>
        <div
          className="second-work d-flex justify-content-center col-5 g-0"
        >
          <img
            alt=""
            className="works-img-service"
            src="static/image/Service.png"
          />
          <div
            className="work-test-1"
          >
            <h4
              className="work-heading"
            >
              Service at your Convenience
            </h4>
            <p
              className="work-para"
            >
              Once we have diagnosed the issue, we fix your
               
              <br />
               
              product, and deliver it to your doorstep â€“ At your
               
              <br />
              Convenience.
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
]
`;
